Rubberduck.Deployment.Build/RubberduckPostBuildTask.cs

```diff
@@ -0,0 +1,365 @@
+using System.Linq;
+using System.Text;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Tasks;
+using Microsoft.Build.Utilities;
+using Microsoft.VisualStudio.Setup.Configuration;
+using Rubberduck.Deployment.Build.Builders;
+using Rubberduck.Deployment.Build.IdlGeneration;
+using Rubberduck.Deployment.Build.Structs;
+using Rubberduck.Deployment.Build.Writers;
+
+namespace Rubberduck.Deployment.Build
+{
+    internal struct DllFileParameters
+    {
+        public string DllFile { get; }
+        public string IdlFile { get; }
+        public string Tlb32File { get; }
+        public string Tlb64File { get; }
+        public string SourceDll { get; }
+        public string TargetDll { get; }
+        public string SourceTlb32 { get; }
+        public string TargetTlb32 { get; }
+        public string SourceTlb64 { get; }
+        public string TargetTlb64 { get; }
+        public string DllXml { get; }
+        public string TlbXml { get; }
+
+        internal DllFileParameters(string dllFile, string sourceDir, string targetDir)
+        {
+            DllFile = dllFile;
+            IdlFile = DllFile.Replace(".dll", ".idl");
+            Tlb32File = DllFile.Replace(".dll", ".x32.tlb");
+            Tlb64File = DllFile.Replace(".dll", ".x64.tlb");
+
+            SourceDll = Path.Combine(sourceDir, DllFile);
+            TargetDll = Path.Combine(targetDir, DllFile);
+            SourceTlb32 = Path.Combine(sourceDir, Tlb32File);
+            TargetTlb32 = Path.Combine(targetDir, Tlb32File);
+            SourceTlb64 = Path.Combine(sourceDir, Tlb64File);
+            TargetTlb64 = Path.Combine(targetDir, Tlb64File);
+            DllXml = TargetDll + ".xml";
+            TlbXml = TargetTlb32 + ".xml";
+        }
+    }
+
+    public class RubberduckPostBuildTask : AppDomainIsolatedTask
+    {
+        [Required]
+        public string Config { get; set; }
+
+        [Required]
+        public string NetToolsDir { get; set; }
+
+        [Required]
+        public string WixToolsDir { get; set; }
+
+        [Required]
+        public string SourceDir { get; set; }
+
+        [Required]
+        public string TargetDir { get; set; }
+
+        [Required]
+        public string ProjectDir { get; set; }
+
+        [Required]
+        public string IncludeDir { get; set; }
+
+        [Required]
+        public string FilesToExtract { get; set; }
+
+        private string RegFilePath =>
+            Path.Combine(Path.Combine(ProjectDir, "LocalRegistryEntries"), "DebugRegistryEntries.reg");
+
+        private string _rootPath;
+        private string _batchPath;
+
+        public override bool Execute()
+        {
+            var result = true;
+            try
+            {
+                this.LogMessage(FormatParameterList());
+
+                CleanOldImports(ProjectDir);
+
+                var dllFiles = FilesToExtract.Split(new[] {"|"}, StringSplitOptions.RemoveEmptyEntries);
+
+                var message = SetVCToolsPath()
```

Comment:
I kinda dislike this state-machine that's falling back to using private fields to transfer information between methods in the same callgraph

```
var useVcTools = TryFindVCTools(out var rootPath, out var batchPath);
LogMessage(useVcTools ? [...] : [...]);
```

feels a little cleaner. Of course this implies a change in `ProcessDll`'s interface to pass root and batch path to it.